odoo/addons/base/wizard/base_partner_merge.py

```diff
@@ -193,10 +193,23 @@ def update_records(model, src, field_model='model', field_id='res_id'):
                 with mute_logger('odoo.sql_db'), self._cr.savepoint():
                     records.sudo().write({field_id: dst_partner.id})
                     records.env.flush_all()
-            except psycopg2.Error:
-                # updating fails, most likely due to a violated unique constraint
-                # keeping record with nonexistent partner_id is useless, better delete it
-                records.sudo().unlink()
+            except psycopg2.Error as e:
+                # updating fails, if the model of records is in allowed ones to delete in case
+                # of unique violation error then delete it, in case is not in allowed ones or
+                # postgresql error is not unique violation, then raise an error
+                model_names_to_delete = self.env['ir.config_parameter'].sudo().get_param("base.models_allowed_to_unlink_partner_merge")
+                model_names_to_delete = model_names_to_delete.split(",") if model_names_to_delete else []
+                if not model_names_to_delete or (records._name in model_names_to_delete and e.pgcode == psycopg2.errorcodes.UNIQUE_VIOLATION):
+                    records.sudo().unlink()
+                elif model_names_to_delete:
+                    # Everything else is not expected and should fail.
+                    msg = """An error has ocurred meanwhile reference fields were updated: \nDestination Record: %s \nSource Record: %s, \nError: %s""" % (
```

Comment:
- Don't use a multiline declaration
- Make it translatable, because this is not only logged but also raised to the user
- "ocurred" -> oc**c**urred

Reply:
>Don't use a multiline declaration

I left the multiline declaration, because I updated the content of msg and the line would be large otherwise.

Other items --> Done